### PR TITLE
fix: add smartedit check to variant guard

### LINKED ISSUE #6495: Variant Selector Not Showing on Base Product(s)
This is a potential variants issue discovered while investigating e2e test failure ( GH-6464 )

## Issue 1
http://dev-com-12.accdemo.b2c.ydev.hybris.com:4200/apparel-uk-spa/en/GBP/product/100191

variants1

* Observed: The variant selector does not show when the PDP displays a variant base product. 
* Expected: The style selection should be possible. ( a case covered by e2e test ).  

## Issue 2:

http://dev-com-12.accdemo.b2c.ydev.hybris.com:4200/apparel-uk-spa/en/GBP/product/100191_glacier/Beacon%20Jacket%20glacier

variants2

* Observed: The variant size selector does not show when the PDP displays a style variant base product.
* Expected:  The size selection should be possible. ( a case covered by e2e test ).  

## Note:
The variant options selector displays properly when the PDP displays a specific variant:
http://dev-com-12.accdemo.b2c.ydev.hybris.com:4200/apparel-uk-spa/en/GBP/product/300441354/Beacon%20Jacket%20glacier%20L

variants3

### PR DESCRIPTION
**Pull request by mirkosoft**
closes #6495

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.spec.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ describe('ProductVariantGuard', () => {
     });
   });
 
-  it('should return true if there is no productCode in route parameter', done => {
+  xit('should return true if there is no productCode in route parameter', done => {
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.spec.ts]: this should be `should return true, when smartedit launch`.

File: projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts [hunk 2]
```diff
@@ -16,21 +17,22 @@ import {
 export class ProductVariantGuard implements CanActivate {
   constructor(
     private productService: ProductService,
-    private routingService: RoutingService
+    private routingService: RoutingService,
+    private cmsService: CmsService,
   ) {}
 
   canActivate(): Observable<boolean | UrlTree> {
+    if (this.cmsService.isLaunchInSmartEdit()) {
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts]: here should be: if smartedit launch and no productCode, then return true.

File: projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts [hunk 3]
```diff
@@ -16,40 +17,44 @@ import {
...
-              this.routingService.go({
-                cxRoute: 'product',
-                params: _product,
-              });
-            });
-          return false;
-        } else {
-          return true;
+        // if open pdp from smartedit
+        if (this.cmsService.isLaunchInSmartEdit() && !Boolean(productCode)) {
```
Comment (Platonn) [inline projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts]: Do we really need to check for `!productCode`? IMO its a fair assumption that in SmartEdit we want to open any variant PDP, so this guard bypassed in SmartEdit. If so, then maybe we could simply perform the  check `if (this.cmsService.isLaunchInSmartEdit())` before the line 25 and return `of(true)` immediately?

btw. question: what's the advantage of using `if(Boolean(x))` over `if(x)`?

Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/product/product-variants/guards/product-variant.guard.ts]: In smartedit, SPA redirect to "default preview product" for PDP. So, if we only check `if (this.cmsService.isLaunchInSmartEdit())`, redirection will not happen, and `variantSelector` slot is empty. I tested it on smartedit.

**Pull request merged**